### ISSUE TITLE
Multiple consecutive savepoints triggered

### ISSUE THREAD

**Issue by pjthepooh** [spotify/flink-on-k8s-operator#115]
I have this weird issue around savepoint. The job runs fine but sometimes it gets  into a state where multiple savepoints will be triggered, about 1 minute each (the job has autoSavepointSeconds: 3600 which works usually fine), see trigger time below
![Screen Shot 2021-09-24 at 6 29 35 PM](https://user-images.githubusercontent.com/12125887/134754761-99301b10-cbfb-4a79-9939-0710434b94a4.png)
After sometime, the job will failed and a a new job will be submitted but can't restore, so it goes into submit-and-fail loop.

I have to re-deploy the resource to put the app restored from the last successful savepoint. However at the beginning of the new job, multiple savepoints will be triggered again, but this time it will mostly be successful. It will stop eventually though, and it becomes normal again. It looks like it tries to complete all the triggered savepoints earlier.
![Screen Shot 2021-09-24 at 7 19 39 PM](https://user-images.githubusercontent.com/12125887/134754866-ec96f20d-cf06-4bcc-9521-8e239c6d4f11.png)

Wonder if anyone encounter this or have any idea. (I'm using the original flink operator from google, but ask here because here is more active)

**Comment by hjwalt** [comment]
@pjthepooh did you encounter a situation where multiple job submitters are created for the same job?

**Comment by pjthepooh** [comment]
No, only one job submitter is created for the same job. I found that this behavior would happen sometimes when restarts occur within the same job, without re-deploying a new flink cluster.

**Comment by pjthepooh** [comment]
From the flinkcluster description, the savepoint creations could be marked as timeout/failed even though they were successfully created shown in the web UI.
```
Warning  SavepointFailed     115s (x666 over 6d5h)    FlinkOperator  Savepoint creation failed: Timed out taking savepoint.
```
